prog.go

```diff
@@ -329,16 +329,6 @@ func newProgramWithOptions(spec *ProgramSpec, opts ProgramOptions, handles *hand
 		return &Program{unix.ByteSliceToString(logBuf), fd, spec.Name, "", spec.Type}, nil
 	}
 
-	// If the caller did not specify a log level,
-	// re-run with branch-level verifier logs enabled.
-	if !opts.LogDisabled && opts.LogLevel == 0 {
-		logBuf = make([]byte, opts.LogSize)
-		attr.LogLevel = LogLevelBranch
-		attr.LogSize = uint32(len(logBuf))
-		attr.LogBuf = sys.NewSlicePointer(logBuf)
-		_, _ = sys.ProgLoad(attr)
-	}
-
 	switch {
 	case errors.Is(err, unix.EPERM):
 		if len(logBuf) > 0 && logBuf[0] == 0 {
```

Comment:
Moving `!opts.LogDisabled && opts.LogLevel == 0` below the switch will make it so that this branch is almost never executed since logBuf is nil in most cases.

It's probably safer to keep the retry where it is.

Reply:
Thanks, missed that.